gum/backend-linux/gumandroid.c

```diff
@@ -666,19 +670,32 @@ gum_try_parse_linker_proc_maps_line (const gchar * line,
 
   if (sizeof (gpointer) == 4)
   {
-    new_path = "/apex/com.android.runtime/bin/linker";
-    old_path = "/system/bin/linker";
+    linker_expected_path = "/system/bin/linker";
   }
   else
   {
-    new_path = "/apex/com.android.runtime/bin/linker64";
-    old_path = "/system/bin/linker64";
+    linker_expected_path = "/system/bin/linker64";
   }
 
-  if (strcmp (path, new_path) == 0)
-    linker_path = new_path;
-  else if (strcmp (path, old_path) == 0)
-    linker_path = old_path;
+  // If there are no linker files, fail
+  if(lstat(linker_expected_path, &linker_stat) == -1) {
+    return FALSE;
+  }
+
+  if (S_ISLNK(linker_stat.st_mode)) {
+    if(readlink(linker_expected_path, linker_true_path, sizeof(linker_true_path)) < 0) {
+      // Issue reading the link, fall back to backwards compatible path
+      // linker_true_path = &linker_expected_path[0];
+      strcpy(linker_true_path, linker_expected_path);
+    }
+  } else {
+    // Expected path is a real file, use it
+    // linker_true_path = &linker_expected_path[0];
+      strcpy(linker_true_path, linker_expected_path);
+  }
```

Comment:
Now that we need to do some I/O inside of this function, which is called for each line of `/proc/self/maps`, we should move this logic outside the function and pass it in instead.

Reply:
Yup - I sort of thought about this, but didn't bother in the moment. Will figure this one out.